core-common/src/main/java/org/mapstruct/factory/Mappers.java

```diff
@@ -119,4 +124,22 @@ private Mappers() {
             throw new RuntimeException( e );
         }
     }
+
+    /**
+     * <p>
+     * Returns the simple name of a class with all enclosing classes' simple names prepended,
+     * separated by an underscore.
+     * </p>
+     * <p>
+     * Example: returns <code>Map_Entry</code> for {@link java.util.Map.Entry}.
+     * </p>
+     */
+    public static String getFlatName(Class<?> klass) {
```

Comment:
Added unit test with two levels.

:100: on making `getFlatName` private (pollutes API), but maybe we don't need it at all (see PR comment)